### DISCUSSION TITLE
llvm + Qt creator + CMake is not working on Windows for me

### DISCUSSION THREAD

**Discussion by moarseniy** [intel/llvm#6297] [Q&A]
Hello, could you help me please, I followed user guide 
https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain-with-support-for-nvidia-cuda 
and successfully installed, configured (with --cuda flag) and compiled llvm. 
It works nice in "x64 Native Tools Command Prompt for VS 2019" on my GPU! 
(It works for me -> 
clang++ -fsycl -fsycl-targets=nvptx64-nvidia-cuda sycl2.cpp -o sycl2.exe)

Then I tried to build and run the same simple VectorAdd example, but using Qt creator and CMake.
I changed the compiler in Qt creator (C and C++) to Clang located in C:\Users\MoArs\sycl_workspace\llvm\build\bin.

my CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.0.0)
project(SYCL_TEST)
add_compile_options(-fsycl -fsycl-targets=nvptx64-nvidia-cuda)
add_executable(SYCL_TEST sycl2.cpp)
target_link_libraries(SYCL_TEST -fsycl -fsycl-targets=nvptx64-nvidia-cuda)
```

CMake configured it but I got the error after attempt to build and run it:

```
error: linker command failed with exit code 1104 (use -v to see invocation)
clang++: error: linker command failed with exit code 1104 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

I would highly appreciate any help!
Thank you

**Comment by bader** [comment]
> target_link_libraries(SYCL_TEST -fsycl -fsycl-targets=nvptx64-nvidia-cuda)

Shouldn't this be `add_link_options(-fsycl -fsycl-targets=nvptx64-nvidia-cuda)` instead?
@intel/llvm-external-read, FYI.

**Reply by moarseniy** [reply]
Should I use something like that?
```
cmake_minimum_required(VERSION 3.0.0)
project(SYCL_TEST)
#add_compile_options(-fsycl -fsycl-targets=nvptx64-nvidia-cuda)
add_executable(SYCL_TEST sycl2.cpp)
add_link_options(-fsycl -fsycl-targets=nvptx64-nvidia-cuda)
```

**Reply by bader** [reply]
You must compile SYCL code with `-fsycl -fsycl-targets=...` options, so do not comment `add_compile_options`.
Also I suggest using target_*_options functions to not interfere with other targets in your cmake files.

**Reply by moarseniy** [reply]
I got it.
So, I haven't commented that but got such error
![image](https://user-images.githubusercontent.com/44135971/173421338-b017998c-0953-473c-b3b5-25db85032e26.png)

**Reply by bader** [reply]
Most likely clang driver picked up wrong STL implementation. I'm not familiar with Qt creator, but is it possible that it configures STL implementation alternative to the provided with MSVC?
